### PR TITLE
CRF: Add scores for decoded tags to crf_decode

### PR DESCRIPTION
**Pull request by tabergma**
Currently, `crf_decode` returns the decoded tags and a best score. The best score contains just the best score of all tags. It does not represent how certain the model is at timestep `x` where it set the tag to `z`. 

This PR updates `crf_decode` to return an additional tensor of size [batch_size, max_seq_len] that contains the score of the decoded tags. This score can be used as confidence score during prediction.

### CODE REVIEW

**Comment by tabergma** [comment]
@googlebot I signed it!

**Review by Squadrick — DISMISSED** [review]
@tabergma Thanks for this contribution. LGTM, sorry for the delay in reviewing this. 

@tensorflow/sig-addons-maintainers Can one of you take a final look and merge this, please?

**Comment by facaiy** [comment]
Thanks, Tanja. I'm wondering if we could add a test case for confidence?  What do you think, Dheeraj @Squadrick

**Comment by tabergma** [comment]
@Squadrick  @facaiy Thanks for your reviews. Should I add a test? What kind of test do you have in mind? Not 100% sure how a test could look like.

File: tensorflow_addons/text/tests/crf_test.py [hunk 1]
```diff
@@ -316,7 +316,7 @@ def test_crf_decode():
         expected_max_sequence = all_sequences[expected_max_sequence_index]
         expected_max_score = all_sequence_scores[expected_max_sequence_index]
 
-        actual_max_sequence, actual_max_score = text.crf_decode(
+        actual_max_sequence, _, actual_max_score = text.crf_decode(
```
Comment (WindQAQ) [inline tensorflow_addons/text/tests/crf_test.py:319]: let's also test if the second return value, say `scores`, is correct.

Comment (tabergma) [reply tensorflow_addons/text/tests/crf_test.py:319]: How would you approach this? In the test we compare the results of `crf_sequence_score` with `crf_decode`. Do I need to update `crf_sequence_score`?

Comment (WindQAQ) [reply tensorflow_addons/text/tests/crf_test.py:319]: Correct!

**Comment by Ghostvv** [comment]
@tabergma any idea why the builds are failing?

**Comment by seanpmorgan** [comment]
Thank you for your contribution. We sincerely apologize for any delay in reviewing, but TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)

**Pull request closed without merging**